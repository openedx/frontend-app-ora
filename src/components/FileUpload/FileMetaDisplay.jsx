import React from 'react';
import PropTypes from 'prop-types';

import { FormattedMessage } from '@edx/frontend-platform/i18n';
import filesize from 'filesize';

import messages from './messages';

const FileMetaDisplay = ({ name, description, size }) => (
  <>
    <div className="file-meta-option">
      <strong><FormattedMessage {...messages.filePopoverNameTitle} /></strong>
      <br />
      {name}
    </div>
    <div className="file-meta-option">
      <strong><FormattedMessage {...messages.filePopoverDescriptionTitle} /></strong>
      <br />
      {description}
    </div>
    <div className="file-meta-option">
      <strong><FormattedMessage {...messages.fileSizeTitle} /></strong>
      <br />
      {typeof (size) === 'number' ? filesize(size) : 'Unknown'}
    </div>
  </>
);

FileMetaDisplay.defaultProps = {
  description: '',
  size: null,
};

FileMetaDisplay.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string,
  size: PropTypes.number,
};

export default FileMetaDisplay;
