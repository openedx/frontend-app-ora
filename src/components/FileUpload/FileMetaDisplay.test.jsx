import { shallow } from '@edx/react-unit-test-utils';

import fileSize from 'filesize';

import FileMetaDisplay from './FileMetaDisplay';

jest.mock('filesize');

describe('<FileMetaDisplay />', () => {
  const props = {
    name: 'name',
    description: 'description',
    size: 123456,
  };
  it('render file meta display', () => {
    fileSize.mockReturnValue('123 KB');
    const wrapper = shallow(<FileMetaDisplay {...props} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });

  it('render file meta display with no size', () => {
    const wrapper = shallow(<FileMetaDisplay {...props} size={null} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });
});
