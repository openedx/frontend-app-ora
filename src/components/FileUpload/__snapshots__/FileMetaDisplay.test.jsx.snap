// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileMetaDisplay /> render file meta display 1`] = `
<Fragment>
  <div
    className="file-meta-option"
  >
    <strong>
      <FormattedMessage
        defaultMessage="File name"
        description="Popover title for file name"
        id="frontend-app-ora.FilePopoverContent.filePopoverNameTitle"
      />
    </strong>
    <br />
    name
  </div>
  <div
    className="file-meta-option"
  >
    <strong>
      <FormattedMessage
        defaultMessage="File description"
        description="Popover title for file description"
        id="frontend-app-ora.FilePopoverCellContent.filePopoverDescriptionTitle"
      />
    </strong>
    <br />
    description
  </div>
  <div
    className="file-meta-option"
  >
    <strong>
      <FormattedMessage
        defaultMessage="File size"
        description=" title for file size"
        id="frontend-app-ora.FileCellContent.fileSizeTitle"
      />
    </strong>
    <br />
    123 KB
  </div>
</Fragment>
`;

exports[`<FileMetaDisplay /> render file meta display with no size 1`] = `
<Fragment>
  <div
    className="file-meta-option"
  >
    <strong>
      <FormattedMessage
        defaultMessage="File name"
        description="Popover title for file name"
        id="frontend-app-ora.FilePopoverContent.filePopoverNameTitle"
      />
    </strong>
    <br />
    name
  </div>
  <div
    className="file-meta-option"
  >
    <strong>
      <FormattedMessage
        defaultMessage="File description"
        description="Popover title for file description"
        id="frontend-app-ora.FilePopoverCellContent.filePopoverDescriptionTitle"
      />
    </strong>
    <br />
    description
  </div>
  <div
    className="file-meta-option"
  >
    <strong>
      <FormattedMessage
        defaultMessage="File size"
        description=" title for file size"
        id="frontend-app-ora.FileCellContent.fileSizeTitle"
      />
    </strong>
    <br />
    Unknown
  </div>
</Fragment>
`;
